library/core/src/cmp.rs

```diff
@@ -578,6 +580,9 @@ impl Ordering {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl StructuralEq for Ordering {}
```

Comment:
Per T-libs (see https://github.com/rust-lang/rust/issues/102371#issuecomment-1261052734) we're going to wait on `derive_const` (or similar) before accepting more changes like this to library.

@rustbot label +S-Blocked